subsys/net/lib/websocket/websocket.c

```diff
@@ -643,10 +656,24 @@ int websocket_recv_msg(int ws_sock, u8_t *buf, size_t buf_len,
 		       u32_t *message_type, u64_t *remaining, s32_t timeout)
 {
 	struct websocket_context *ctx;
-	size_t header_len, pos_to_write = 0;
+	size_t header_len = 0, pos_to_write = 0;
 	int recv_len = 0;
 	int ret;
 
+#if defined(CONFIG_NET_TEST)
+	/* Websocket unit test does not use socket layer but feeds
```

Comment:
I would image we'd end up with such "test handler" in the code eventually, e.g. for fault injection. But is it warranted here now? What tests you can't perform via socket layer which required such contraptions? In the end, users will user this API via socket layer, and that's what we're expected to test.

Reply:
I do not want to implement websocket server just to test this receive function. Using sockets would be nice of course but here sockets just complicate things unnecessarily. It should not really matter where the input data comes from to this function when testing it.